### PR TITLE
Fix #88606 issue about wrong warning message in kubeadm

### LINKED ISSUE #88606: Incorrect warning message output when using apiserver extra options (authorization-mode)
**What happened**:

when i installed k8s using kubeadm and my custom config like
```
apiVersion: kubeadm.k8s.io/v1beta2
kind: ClusterConfiguration
kubernetesVersion: 1.17.3
apiServer:  
  extraArgs:    
    authorization-mode: "Node,RBAC"
```

then output is

```
W0227 07:14:10.425558   16029 manifests.go:214] the default kube-apiserver authorization-mode is "Node,RBAC"; using "Node,RBAC"
```

It is a bug that the default value(Node,RBAC) is entered but a warning message is displayed.

https://github.com/kubernetes/kubernetes/blob/4e79344501bfc5d3bc48caa12c41b9246d90850d/cmd/kubeadm/app/phases/controlplane/manifests.go#L207

A function must be added to compare two string slices.

**What you expected to happen**:

If the default value and custom value are the same, no warning message should be displayed.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
1.17.3
- Install tools:
kubeadm

/area kubeadm
/feature bug
/sig cluster-lifecycle

### PR DESCRIPTION
**Pull request by seongenie**
**What type of PR is this?**
> /kind bug
> /assign @timothysc

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #88606 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cmd/kubeadm/app/phases/controlplane/manifests.go [hunk 1]
```diff
@@ -222,16 +223,25 @@ func getAuthzModes(authzModeExtraArgs string) string {
...
 		}
 	}
 	return strings.Join(defaultMode, ",")
 }
 
+// compare two modes which contents are same regardless of index.
+func compareMode(modeA, modeB []string) bool {
+	sort.Strings(modeA)
+	sort.Strings(modeB)
+	return strings.Join(modeA, ",") == strings.Join(modeB, ",")
```
Comment (zhouya0) [inline cmd/kubeadm/app/phases/controlplane/manifests.go]: Thanks for your PR! I concern about strings.Join() would be called three times:
1. The `compareMode` func
2. klog
3. func `getAuthzModes` return
I think it's better to optimize this.

Comment (neolit123) [reply cmd/kubeadm/app/phases/controlplane/manifests.go]: i've seen the warning myself and i acknowledge that this should be fixed.
but the ordering of auth modes for the API server is important, so there should be no sorting!

the compare function should look like this:
```
// compareAuthzModes compares two given authz modes and returns false if they do not match
func compareAuthzModes(a, b []string) bool {
	if len(a) != len(b) {
		return false
	}
	for i, m := range a {
		if m != b[i] {
			return false
		}
	}
	return true
}
```
@seongenie please update the PR to use that, and also use `git commit -a` and `git push -f` to keep the number of commits to 1.

thanks

Comment (neolit123) [reply cmd/kubeadm/app/phases/controlplane/manifests.go]: you can also use:
`./hack/verify-gofmt.sh` to make sure the code is formatted correctly.

File: cmd/kubeadm/app/phases/controlplane/manifests.go [hunk 2]
```diff
@@ -22,6 +22,7 @@ import (
 	"path/filepath"
 	"strconv"
 	"strings"
+	"sort"
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/controlplane/manifests.go]: sort is no longer needed.

**Comment by neolit123** [comment]
please squash the commits to 1.

**Comment by neolit123** [comment]
@seongenie 
> please squash the commits to 1.

^ please let me know if you need help with that.

a mini guide would be:
- do a `git rebase -i HEAD~3` on your local branch then add "fixup" two the two secondary commits 
- force push the local change to your remote branch using `git push -f <remote-name> <branch-name>`

**Comment by neolit123** [comment]
same PR is here:
https://github.com/kubernetes/kubernetes/pull/90064

thanks
/close

**Pull request closed without merging**